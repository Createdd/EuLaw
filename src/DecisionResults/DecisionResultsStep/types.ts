interface DecisionResultsStepPropsType {
    //
}

interface DecisionResultsStepStateType {
    isGoing: boolean;
    numberOfGuests: number;
}

export {
    DecisionResultsStepPropsType,
    DecisionResultsStepStateType,
};
