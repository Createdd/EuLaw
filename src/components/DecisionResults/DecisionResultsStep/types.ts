interface DecisionResultsStepPropsType {
    //
}

interface DecisionResultsStepStateType {
    checked: boolean;
}

export {
    DecisionResultsStepPropsType,
    DecisionResultsStepStateType,
};
