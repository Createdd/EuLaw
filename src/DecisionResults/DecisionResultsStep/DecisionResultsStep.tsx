import * as React from 'react';

import {
  DecisionResultsStepPropsType,
  DecisionResultsStepStateType
} from './types';
import DecisionResultsStepStyled from './styles';

class DecisionResultsStep extends React.Component<
  DecisionResultsStepPropsType,
  DecisionResultsStepStateType
> {
  constructor(props: DecisionResultsStepPropsType) {
    super(props);
    this.state = {
      isGoing: true,
      numberOfGuests: 2
    };
  }

  /* tslint:disable no-any */
  handleInputChange = (event: any) => {
    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    this.setState({
      [name]: value
    });
  }

  render() {
    return (
      <DecisionResultsStepStyled>
        <label>
          <input
            name="isMemberState"
            type="checkbox"
            checked={this.state.isGoing}
            onChange={this.handleInputChange}
            disabled={true}
          />
          is a member state
        </label>
      </DecisionResultsStepStyled>
    );
  }
}

export { DecisionResultsStep as default };
