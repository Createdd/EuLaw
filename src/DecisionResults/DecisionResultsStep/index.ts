export { default } from './DecisionResultsStep';
export {
    DecisionResultsStepPropsType,
    DecisionResultsStepStateType,
} from './types';
