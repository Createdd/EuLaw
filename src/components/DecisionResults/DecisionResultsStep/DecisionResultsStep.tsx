import * as React from 'react';

import {
  DecisionResultsStepPropsType,
  DecisionResultsStepStateType
} from './types';
import DecisionResultsStepStyled from './styles';

class DecisionResultsStep extends React.Component<
  DecisionResultsStepPropsType,
  DecisionResultsStepStateType
> {
  constructor(props: DecisionResultsStepPropsType) {
    super(props);
    this.state = {
      checked: false
    };
  }

  // copied from React Docs
  handleInputChange = (event: React.FormEvent<HTMLInputElement>): void => {
    // const target = event.target as HTMLInputElement;
    // const value = target.type === 'checkbox' ? target.checked : target.value;
    // const name = target.name;

    this.setState({
      checked: true
    });
  }

  render() {
    return (
      <DecisionResultsStepStyled>
        <label>
          <input
            name="isMemberState"
            type="checkbox"
            checked={true}
            onChange={this.handleInputChange}
            disabled={true}
          />
          is a member state
        </label>
      </DecisionResultsStepStyled>
    );
  }
}

export { DecisionResultsStep as default };
