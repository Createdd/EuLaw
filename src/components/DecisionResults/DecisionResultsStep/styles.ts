import styled from 'styled-components';

const DecisionResultsStepStyled = styled.div``;

const StyledDecisionStepSuccess = styled.li`
  color: green;
`;

const StyledDecisionStepFail = styled.li`
  color: red;
`;

export {
  DecisionResultsStepStyled,
  StyledDecisionStepSuccess,
  StyledDecisionStepFail
};
