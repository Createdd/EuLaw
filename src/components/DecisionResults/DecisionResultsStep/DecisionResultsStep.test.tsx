import * as React from 'react';
import * as renderer from 'react-test-renderer';

import DecisionResultsStep from './DecisionResultsStep';

const testResults = {
    topics: ['test1', 'test2'],
    places: ['test1', 'test2']
  };

describe( 'Test DecisionResultsStep', () => {
    it( 'renders snapshot', () => {
        const tree = renderer.create(
            <DecisionResultsStep results={testResults} success={true} type="MS"/>
        ).toJSON();
        
        expect( tree ).toMatchSnapshot();
    } );
} );
