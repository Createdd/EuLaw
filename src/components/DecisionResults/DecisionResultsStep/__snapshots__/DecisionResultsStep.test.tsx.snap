// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DecisionResultsStep renders snapshot 1`] = `
<div
  className="sc-bdVaJa bDWFJH"
>
  <li
    className="sc-bwzfXH kuSbvP"
  >
    <label>
      Is a 
      MS
    </label>
    <input
      checked={true}
      disabled={true}
      name="MS"
      type="checkbox"
    />
  </li>
</div>
`;
